### PR TITLE
Commands Rewrite

### PR DESCRIPTION
**Pull request by Sxtanna**
This PR contains rewrites of the commands using ACF

### CODE REVIEW

**Comment by Sxtanna** [comment]
WIP btw

File: src/main/kotlin/Hound.kt [hunk 1]
```diff
@@ -40,19 +43,134 @@ class Hound : JavaPlugin() {
...
+                origin.toBigDecimal().setScale(2, RoundingMode.HALF_EVEN).toDouble() + (input.drop(1).toDoubleOrNull() ?: 0.0)
+            }
+        }
+        
+        manager.registerCommand(cmds.BlockSearchCommand(this))
+        manager.registerCommand(cmds.ChestSearchCommand(this))
+        manager.registerCommand(cmds.LiveChestSearchCommand(this))
+        manager.registerCommand(cmds.TargetCommand(this))
+        manager.registerCommand(cmds.TorchGuideCommand(this))
+
```
Comment (vinhowe) [inline src/main/kotlin/Hound.kt:172]:
```suggestion
```

File: src/main/kotlin/Hound.kt [hunk 2]
```diff
@@ -40,19 +43,134 @@ class Hound : JavaPlugin() {
...
+                    context.hasFlag("y") -> player.location.y
+                    context.hasFlag("z") -> player.location.z
+                    else -> {
+                        throw InvalidCommandArgument("&4Coordinate must be absolute.")
+                    }
+                }
+
+                origin.toBigDecimal().setScale(2, RoundingMode.HALF_EVEN).toDouble() + (input.drop(1).toDoubleOrNull() ?: 0.0)
+            }
+        }
```
Comment (vinhowe) [inline src/main/kotlin/Hound.kt:165]: It feels to me like there are an awful lot of linebreaks here.

File: src/main/kotlin/cmds/BlockSearchCommand.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		
+		val blocks = hound.highlightBlockTypeForPlayer(material, player)
+		
+		
+		if (blocks.isNotEmpty()) {
+			player.sendMessage("§6Found ${blocks.size} ${if (blocks.size == 1) "block" else "blocks"} of type '${material.name.toLowerCase()}'")
+		} else {
+			player.sendMessage("§4Couldn't find any blocks of type '${material.name.toLowerCase()}' within $radius ${if (radius == 1) "block" else "blocks"} of you.")
+		}
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/BlockSearchCommand.kt:34]: Again, there are a lot of empty lines here. It just feels weird I guess.

File: src/main/kotlin/cmds/ChestSearchCommand.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import PartialMatchingContainersSearchResult
+import co.aikar.commands.BaseCommand
+import co.aikar.commands.annotation.*
+import data.PartialMatchModel
+import org.bukkit.entity.Player
+
+@CommandAlias("chound|cf")
+class ChestSearchCommand(private val hound: Hound) : BaseCommand() {
+
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/ChestSearchCommand.kt:14]:
```suggestion
```

File: src/main/kotlin/cmds/ChestSearchCommand.kt [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                "§4Couldn't find items matching '${model.input}' in a container within $radius ${if (radius == 1) "block" else "blocks"} of you."
+            }
+        }
+
+        player.sendMessage(response)
+    }
+
+
+    private fun String.plural(count: Int, suffix: String) = if (count == 1) this else "$this$suffix"
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/ChestSearchCommand.kt:47]:
```suggestion
    private fun String.plural(count: Int, suffix: String) = if (count == 1) this else "$this$suffix"
```

Some inconsistent 2 line breaks. Correct me if this is a Kotlin convention or something.

File: src/main/kotlin/cmds/ChestSearchCommand.kt [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+class ChestSearchCommand(private val hound: Hound) : BaseCommand() {
+
+
+    @Default
+    @Syntax("<item>")
+    @CommandCompletion("@items")
+    @CommandPermission("hound.search.container")
+    @Description("Highlight item in nearby containers.")
+    fun search(player: Player, model: PartialMatchModel) {
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/ChestSearchCommand.kt:21]:
```suggestion
```

File: src/main/kotlin/cmds/ChestSearchCommand.kt [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                }
+            }
+
+            if (hasFuzzy) {
+                append("$fuzzyMatchCount partial ${"match".plural(fuzzyMatchCount, "es")}")
+            }
+
+            append(" for '$displayName' in $totalChestCount ${"chest".plural(totalChestCount, "s")}")
+        }
+    }
```
Comment (vinhowe) [inline src/main/kotlin/cmds/ChestSearchCommand.kt:83]: These both look fantastic. A big improvement.

File: src/main/kotlin/cmds/LiveChestSearchCommand.kt [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+            hound.liveSearchSet -= player.uniqueId
+            hound.clearStaticHighlightsForPlayer(player)
+
+            player.sendMessage("§6Deactivated live search mode.")
+        }
+
+    }
+
+}
```
Comment (vinhowe) [inline src/main/kotlin/cmds/LiveChestSearchCommand.kt:40]:
```suggestion
        }
    }
}
```

File: src/main/kotlin/cmds/LiveChestSearchCommand.kt [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+import Hound
+import co.aikar.commands.BaseCommand
+import co.aikar.commands.annotation.*
+import data.LiveSearchState
+import org.bukkit.entity.Player
+
+@CommandAlias("lchound|lcf")
+class LiveChestSearchCommand(private val hound: Hound) : BaseCommand() {
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/LiveChestSearchCommand.kt:11]:
```suggestion
```

File: src/main/kotlin/cmds/TargetCommand.kt [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    @CommandPermission("hound.target")
+    @Description("Create a floating guide that will lead you to coordinates.")
+    fun target(player: Player, @Flags("x") @Optional x: Double?, @Flags("z") @Optional z: Double?) {
+
+        if (x == null || z == null) {
+            hound.clearGuideForPlayer(player)
+        } else {
+            hound.createGuideForPlayer(player, x, z)
+        }
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/TargetCommand.kt:23]:
```suggestion
        if (x == null || z == null) {
            hound.clearGuideForPlayer(player)
        } else {
            hound.createGuideForPlayer(player, x, z)
        }
```

File: src/main/kotlin/cmds/TorchGuideCommand.kt [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+    @CommandPermission("hound.torchguide")
+    @Description("Highlight a grid guide for torches")
+    fun guide(player: Player, @Optional spacing: Int?) {
+
+        if (spacing == null) {
+            hound.clearStaticHighlightsForPlayer(player)
+        } else {
+            hound.createTorchHighlightsForPlayer(player, spacing)
+        }
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/TorchGuideCommand.kt:23]:
```suggestion
        if (spacing == null) {
            hound.clearStaticHighlightsForPlayer(player)
        } else {
            hound.createTorchHighlightsForPlayer(player, spacing)
        }
```

File: src/main/kotlin/cmds/BlockSearchCommand.kt [hunk 12]
```diff
@@ -0,0 +1,37 @@
+package cmds
```
Comment (vinhowe) [inline src/main/kotlin/cmds/BlockSearchCommand.kt:1]: Could we go with either `commands` or `command` for the package name, depending on convention? This feels unnecessarily succinct.

File: src/main/kotlin/Hound.kt [hunk 13]
```diff
@@ -40,19 +43,134 @@ class Hound : JavaPlugin() {
...
+                    else -> Double.NaN
+                }
+
+                if (!origin.isNaN()) {
+                    suggestions += origin.toBigDecimal().setScale(2, RoundingMode.HALF_EVEN).toPlainString()
+                }
+            }
+
+            suggestions
+        }
```
Comment (vinhowe) [inline src/main/kotlin/Hound.kt:84]: These completions make the command feel a lot more like a built-in command (like `tp`). Good job.

File: src/main/kotlin/cmds/LiveChestSearchCommand.kt [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    @Default
+    @Syntax("<on/off>")
+    @CommandCompletion("@states")
+    @CommandPermission("hound.search.container.live")
+    @Description("Live search mode. While active, highlights item held in main hand in nearby containers.")
+    fun search(player: Player, state: LiveSearchState) {
+
+        if (state.currentState == state.desiredState) {
+            return player.sendMessage("§4Live search mode is already ${if (state.currentState) "active" else "inactive"}.")
+        }
```
Comment (vinhowe) [inline src/main/kotlin/cmds/LiveChestSearchCommand.kt:22]: I might be missing something here, but why can't you throw a command error in Hound.kit in `registerIssuerAwareContext(LiveSearchState::class.java)` if current state == desired state?

File: src/main/kotlin/cmds/LiveChestSearchCommand.kt [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+        if (state.desiredState) {
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/LiveChestSearchCommand.kt:25]:
```suggestion
```

File: src/main/kotlin/cmds/LiveChestSearchCommand.kt [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+            hound.liveSearchSet += player.uniqueId
+            hound.highlightItemTypeForPlayer(player.inventory.itemInMainHand.type, player)
+
+            player.sendMessage("§6Activated live search mode.")
+        } else {
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/LiveChestSearchCommand.kt:31]:
```suggestion
```

File: src/main/kotlin/cmds/ChestSearchCommand.kt [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+
```
Comment (vinhowe) [inline src/main/kotlin/cmds/ChestSearchCommand.kt:84]:
```suggestion
```

**Review by vinhowe — CHANGES_REQUESTED** [review]
Mostly just whitespace but I had a few questions too. Let me know if I missed anything you wanted me to look at.

**Review by vinhowe — APPROVED** [review]
LGTM

**Pull request merged**